addons/stock/models/stock_move_line.py

```diff
@@ -142,7 +142,7 @@ def _compute_quantity(self):
         for record in self:
             if not record.quant_id or record.quantity:
                 continue
-            if float_compare(record.move_id.product_qty, record.quantity, precision_rounding=record.move_id.product_uom.rounding) > 0:
+            if float_compare(record.move_id.product_qty, record.quantity, precision_rounding=record.product_uom_id.rounding) > 0:
```

Comment:
Seems weird to me to do that check if there is no move in the first place. I would assume that we'd want to fallback on the `else` in that case.

Reply:
Also, looking at it, the whole `if` block looks kinda suspicious, as we compare `move_id.product_qty` (which are in the **product's uom**) with `move_id.quantity` (which are in the **move's uom**). Would be great to rewrite that so it correctly uses the right uoms.